### ISSUE TITLE
Build failed when put in appcenter

### ISSUE THREAD

**Issue by xylim666** [hanabi1224/flutter_appcenter_bundle#5]
Below error when building in appcenter

Execution failed for task ':flutter_appcenter_bundle:compileReleaseKotlin'.
> Could not resolve all artifacts for configuration ':flutter_appcenter_bundle:releaseCompileClasspath'.

   > Failed to transform file 'flutter.jar' to match attributes {artifactType=android-classes, org.gradle.usage=java-runtime-jars}

      > Execution failed for JetifyTransform: /Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar.

         > Failed to transform '/Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar' using Jetifier. Reason: Cannot open a library at 'FileMapping(from=/Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar, to=/Users/runner/.gradle/caches/transforms-2/files-2.1/1aa288544520ad5336f0b9b18fb858af/jetified-flutter.jar)'. (Run with --stacktrace for more details.)

**Comment by Vayth** [comment]
I have similar experience with appcenter. For flutter app.
My workaround for now, and for dealing with flutter at the moment, is building in other place (Azure Devops in my case), then push it to appcenter for distribution and analytics.

I'm not sure whether the problem lies with the appcenter itself or this package build setting, although I hope the dev could look into it as well.

**Comment by dinovo** [comment]
Facing the same problem as described while trying to build on AppCenter.

**Comment by hanabi1224** [comment]
I don't have a repo environment but I've tried to make a fix with the error message.  Please try `flutter_appcenter_bundle: ^1.0.0-dev+4` and let me know if it works

**Comment by Vayth** [comment]
Unfortunately, it's the same for me.
For Android:

```
FAILURE: Build failed with an exception.



* What went wrong:

Execution failed for task ':flutter_appcenter_bundle:compileReleaseKotlin'.

> Could not resolve all artifacts for configuration ':flutter_appcenter_bundle:releaseCompileClasspath'.

   > Failed to transform file 'flutter.jar' to match attributes {artifactType=android-classes, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}

      > Execution failed for JetifyTransform: /Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar.

         > Failed to transform '/Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar' using Jetifier. Reason: Cannot open a library at 'FileMapping(from=/Users/runner/runners/2.164.6/work/1/s/flutter/bin/cache/artifacts/engine/android-x64/flutter.jar, to=/Users/runner/.gradle/caches/transforms-2/files-2.1/7b710bd06c3f62ec4ea976088fe76f1d/jetified-flutter.jar)'. (Run with --stacktrace for more details.)


```

As for ios:

```
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.0.0+3/ios/Classes/FlutterAppcenterBundlePlugin.m:8:9: fatal error: 'flutter_appcenter_bundle-Swift.h' file not found
    #import "flutter_appcenter_bundle-Swift.h"
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description
```

**Comment by Vayth** [comment]
Upgrading to 1.1.0+1, android built successfully.

iOS still have the same issue.

```
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppcenterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/runner/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/FlutterAppcenterBundlePlugin.m:8:9: fatal error: 'flutter_appcenter_bundle-Swift.h' file not found
    #import "flutter_appcenter_bundle-Swift.h"
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description
```

**Comment by hanabi1224** [comment]
@Vayth  Hmm, iOS build errors can be tricky. Can you please provide a minimal repro with your build environment info

**Comment by Vayth** [comment]
It's actually a new project created by flutter cli (counter increment app) with setup guided by [this site](https://buildflutter.com/deploying-flutter-apps-via-appcenter/).
The initial build built successfully, but got this error once I added your package and rebuilt it again

**Comment by dinovo** [comment]
Upgrading to 1.1.0+1 resolved my Android build issue, it now builds successfully in AppCenter. I've also built my iOS project to make sure it didn't break and that also succeeded in AppCenter.

Thanks for fixing the issue! 👍

**Comment by xylim666** [comment]
Thanks, 1.1.0+1 fixed it for me both ios and android

**Comment by marqueeeeeee** [comment]
not working for iOS, im in version 1.1.0+1

`/Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:20:9: warning: result of call to 'register(binaryMessenger:)' is unused
            register(binaryMessenger: registrar.messenger());
            ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:46: warning: treating a forced downcast to 'String' as optional will never produce 'nil'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                 ^
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:48: note: use 'as?' to perform a conditional downcast to 'String'
                MSAppCenter.start(call.arguments as! String, withServices:[
                                                   ^
                                                   ?
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/SwiftFlutterAppce
    nterBundlePlugin.swift:27:31: note: add parentheses around the cast to silence this warning
                MSAppCenter.start(call.arguments as! String, withServices:[
                                  ^
                                  (                        )
    /Users/mark/Documents/flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_appcenter_bundle-1.1.0+1/ios/Classes/FlutterAppcenterB
    undlePlugin.m:8:9: fatal error: 'flutter_appcenter_bundle-Swift.h' file not found
    #import "flutter_appcenter_bundle-Swift.h"
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description`

**Comment by dustincatap** [comment]
Hi, I'm facing the iOS issue as well.

Using version 1.1.0+1

```
FlutterAppcenterBundlePlugin.m:8:9: fatal error: 'flutter_appcenter_bundle-Swift.h' file not found
               #import "flutter_appcenter_bundle-Swift.h"
```

The build fails when building using Azure Pipelines. But when I'm using my local Mac, it builds successfully 😕

**Comment by marqueeeeeee** [comment]
Im having this issue as well when building using codemagic.io

**Comment by Vayth** [comment]
> Thanks, 1.1.0+1 fixed it for me both ios and android

Did you make some specific configuration on ios/appcenter?

**Comment by hanabi1224** [comment]
No iOS setup is just from standard plugin template.  Please reference CI setup which builds successfully for iOS

**Comment by dustincatap** [comment]
I've cloned this project and trying to replicate this error using my workstation (macOS 10.14.6, Xcode 11.3.1). It seems that it is intermittent, I only managed to replicate it once after many builds. Will update you if I find any.

**Comment by dustincatap** [comment]
After some searching, I found these issues [#25676](https://github.com/flutter/flutter/issues/25676), [#24099](https://github.com/flutter/flutter/issues/24099) which are connected to this issue in iOS. It seems that there is a problem in using plugins that are written in Swift.

As a quick fix, I edited the `Podfile` in my iOS Runner project and commented the `use_frameworks!` part, and the project successfully builds in my CI pipeline. The only thing I don't understand is that my project builds in my workstation without editing the `Podfile` 😕
